core/src/test/java/io/aiven/kafka/tieredstorage/chunkmanager/cache/DiskBasedChunkCacheMetricsTest.java

```diff
@@ -0,0 +1,131 @@
+
+import org.apache.kafka.common.TopicIdPartition;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.Uuid;
+import org.apache.kafka.common.metrics.MetricConfig;
+import org.apache.kafka.server.log.remote.storage.RemoteLogSegmentId;
+import org.apache.kafka.server.log.remote.storage.RemoteLogSegmentMetadata;
+
+import io.aiven.kafka.tieredstorage.chunkmanager.ChunkManager;
+import io.aiven.kafka.tieredstorage.manifest.SegmentManifest;
+
+import org.assertj.core.data.Percentage;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class DiskBasedChunkCacheMetricsTest {
+    static final MBeanServer MBEAN_SERVER = ManagementFactory.getPlatformMBeanServer();
+
+    static final long METRIC_TIME_WINDOW_SEC =
+        TimeUnit.SECONDS.convert(new MetricConfig().timeWindowMs(), TimeUnit.MILLISECONDS);
+
+    public static final Uuid SEGMENT_ID = Uuid.randomUuid();
+
+    static final int LOG_SEGMENT_BYTES = 10;
+    static final RemoteLogSegmentMetadata REMOTE_LOG_SEGMENT_METADATA =
+        new RemoteLogSegmentMetadata(
+            new RemoteLogSegmentId(
+                new TopicIdPartition(Uuid.randomUuid(), new TopicPartition("topic", 0)),
+                SEGMENT_ID),
+            1, -1, -1, -1, 1L,
+            LOG_SEGMENT_BYTES, Collections.singletonMap(1, 100L));
+
+    @Mock
+    ChunkManager chunkManager;
+    @Mock
+    SegmentManifest segmentManifest;
+    @TempDir
+    Path baseCachePath;
+
+    DiskBasedChunkCache diskBasedChunkCache;
+
+    @BeforeEach
+    void setUp() {
+        diskBasedChunkCache = new DiskBasedChunkCache(chunkManager);
+        diskBasedChunkCache.configure(Map.of(
+            "retention.ms", "-1",
+            "size", "-1",
+            "path", baseCachePath.toString()
+        ));
+    }
+
+    @Test
+    void shouldRecordMetrics() throws Exception {
+        when(chunkManager.getChunk(any(), eq(segmentManifest), eq(0)))
+            .thenReturn(new ByteArrayInputStream("test".getBytes()));
+
+        diskBasedChunkCache.getChunk(REMOTE_LOG_SEGMENT_METADATA, segmentManifest, 0);
+        diskBasedChunkCache.getChunk(REMOTE_LOG_SEGMENT_METADATA, segmentManifest, 0);
+
+        final var objectName = new ObjectName("aiven.kafka.server.tieredstorage.cache:type=chunk-cache");
+        assertThat(MBEAN_SERVER.getAttribute(objectName, "cache-hits-total"))
+            .isEqualTo(1.0);
+        assertThat((double) MBEAN_SERVER.getAttribute(objectName, "cache-hits-rate"))
+            .isCloseTo(1.0 / METRIC_TIME_WINDOW_SEC, Percentage.withPercentage(99));
```

Comment:
Instead of all these class casts to `double` that could lead to potential NPE in case of null, should we use this?
```suggestion
        assertThat(MBEAN_SERVER.getAttribute(objectName, "cache-hits-rate"))
            .asInstanceOf(InstanceOfAssertFactories.DOUBLE)
            .isCloseTo(1.0 / METRIC_TIME_WINDOW_SEC, Percentage.withPercentage(99));
```
Might be a little bit too verbose however.

Reply:
Agree, it may be too verbose. We may have used double casting elsewhere. Would prefer to keep it if that's the case